src/Magnum/GL/Implementation/ShaderState.h

```diff
@@ -53,6 +53,7 @@ struct ShaderState {
 
     void(Shader::*addSourceImplementation)(std::string);
     void(*cleanLogImplementation)(std::string&);
+    void(*completionStatusImplementation)(GLuint, GLenum, GLint* value);
```

Comment:
To fix the 32bit Windows build, this should be enough I hope -- it's just copied from another `State.h` header:

```suggestion
    /* This is a direct pointer to a GL function, so needs a __stdcall on
       Windows to compile properly on 32 bits */
    void(APIENTRY *completionStatusImplementation)(GLuint, GLenum, GLint* value);
```

Same would be in `ShaderProgramState.h`, and the fallback function declarations in `GL/Shader.h` and `GL/AbstractShaderProgram.h` would then need `APIENTRY` as well -- hopefully this is the right place to put it:

```cpp
static MAGNUM_GL_LOCAL APIENTRY void completionStatusImplementationFallback(GLuint, GLenum, GLint*);
```

I can also do this myself during merge if you don't find time in the next days.